daml-assistant/daml-helper/test/DA/Daml/Helper/Test/Ledger.hs

```diff
@@ -0,0 +1,55 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Daml.Helper.Test.Ledger (main) where
+
+import DA.Bazel.Runfiles
+import DA.Ledger.Services.PartyManagementService (PartyDetails(..))
+import DA.Ledger.Types (Party(..))
+import DA.Test.HttpJson
+import DA.Test.Sandbox
+import System.Environment.Blank
+import System.FilePath
+import System.Process
+import Test.Tasty
+import Test.Tasty.HUnit
+
+main :: IO ()
+main = do
+  setEnv "TASTY_NUM_THREADS" "1" True
+  damlHelper <-
+    locateRunfiles (mainWorkspace </> "daml-assistant" </> "daml-helper" </> exe "daml-helper")
+  defaultMain $
+    testGroup
+      "daml ledger"
+      [ withSandbox defaultSandboxConf $ \getSandboxPort -> do
+          withHttpJson getSandboxPort $ \getHttpJson ->
+            testGroup
+              "list-parties"
+              [ testCase "succeeds against HTTP JSON API" $ do
+                  HttpJson {hjPort, hjTokenFile} <- getHttpJson
+                  sandboxPort <- getSandboxPort
+                  callCommand $
+                    unwords
+                      [ damlHelper
+                      , "ledger"
+                      , "allocate-party"
+                      , "--host"
+                      , "localhost"
+                      , "--port"
+                      , show sandboxPort
+                      , "alice"
+                      ]
+                  let ledgerOpts =
+                        [ "--host=localhost"
+                        , "--json-api"
+                        , "--port"
+                        , show hjPort
+                        , "--access-token-file"
+                        , hjTokenFile
+                        ]
+                  out <- readProcess damlHelper ("ledger" : "list-parties" : ledgerOpts) ""
+                  ((show $ PartyDetails (Party "alice") "alice" True) `elem` lines out) @?
```

Comment:
Does `list-parties` use `show` to display its output rather than render it to JSON?

Reply:
yes, unless you add the `--json` flag to the command.

Reply:
Oh, wow. That's not particularly pretty. I'm wondering if it would make sense to make JSON pretty-printing the default when running commands against the JSON API. In fact, we could just pretty print the answer we receive from the JSON API. We should probably also revisit the default for running the commands against the GRCP API. @cocreature What do you think? @bame-da How much would you be opposed to changing such a default?

Reply:
Definitely in favor of changing that. I hate the `show` format.

Reply:
Also agree with changing this. But this should be part of a different PR.

Reply:
It would be nice if the output didn't depend on the API you hit.

Reply:
@bame-da that's already the case.

Reply:
@bame-da Does that mean you would be ok if output things as JSON by default, regardless of the API we hit, and provide a flag to enable the current "legacy output format" (which uses Haskell's show) for users who mysteriously rely on that format?

Reply:
Yes, I'd be OK with that.

Reply:
@hurryabit . I'll work on it soon.

Reply:
If I understood @bame-da correctly today, he has withdrawn his support for changing the default output format to JSON.

Reply:
@hurryabit @bame-da Could you please put down the final decision here?

Reply:
My original comment was based on the assumption that we were in an inconsistent state currently, and that the output depended on the API we were hitting:

> It would be nice if the output didn't depend on the API you hit.

Assuming we had been in such a bad state, I would have been OK to correct it towards the better version, which is consistently emitting the JSON version. However, we seem to be in a reasonable state with two completely orthogonal flags ``--json-api`` and ``--json``. I don't see a strong case here for making a backwards-breaking change switching the default to ``--json``.

Reply:
Let's not do it now then and wait until enough customers have asked us for JSON output.

Reply:
We already have json output, it’s just not the default.

Reply:
I know. But how many people do read help texts or documentation? If it's not the default, it's not there. :trollface: